src/main/webapp/populateData.js

```diff
@@ -15,8 +15,14 @@
 // Script to handle populating data in the panels
 
 /* eslint-disable no-unused-vars */
-/* global signOut, AuthenticationError */
+/* global signOut, AuthenticationError, getDateInLocalTimeZone */
 // TODO: Refactor so populate functions are done in parallel (Issue #26)
+
+// Stores the last retrieved copy of the user's tasklists and tasks
+// (mapped by tasklistId)
+let tasklists = [];
+let tasks = {};
```

Comment:
Would it be better to have tasklists as {} and have each tasklist map to a [] of tasks?

Reply:
Really interesting question actually - so this is possible in Java with hashmaps, but its impossible in JavaScript with objects. 

Objects need to have a key that's a string - not an entire object. I could build a custom data structure that lets you get the tasks from a specific tasklist, but under the hood it would essentially boil down to this. 

I'll probably take a look at making this cleaner later (maybe even building that data structure), but for now I just wanted to demonstrate that the server side changes work by implementing a simple implementation on the client. 

Open for suggestions though if you have an easy way to associate objects with a list of objects in JS

Reply:
Ok cool! I agree that it's probably going to be [less clean](https://stackoverflow.com/questions/36299664/javascript-convert-a-list-of-objects-to-an-associative-array) if you were to associate the objects with a list of objects in JS.